operators/advanced_network/python/check_system.py

```diff
@@ -0,0 +1,653 @@
+                and sm_max is not None
+                and mem_current is not None
+                and mem_max is not None
+            ):
+                logging.info(f"GPU {current_gpu}: Checking clocks...")
+
+                if sm_current != sm_max:
+                    logging.warning(
+                        f"GPU {current_gpu}: SM Clock is set to {sm_current} MHz, but should be {sm_max} MHz."
+                    )
+                else:
+                    logging.info(f"GPU {current_gpu}: SM Clock is correctly set to {sm_max} MHz.")
+
+                if mem_current != mem_max:
+                    logging.warning(
+                        f"GPU {current_gpu}: Memory Clock is set to {mem_current} MHz, but should be {mem_max} MHz."
+                    )
+                else:
+                    logging.info(
+                        f"GPU {current_gpu}: Memory Clock is correctly set to {mem_max} MHz."
+                    )
+
+                # Reset variables for the next GPU section
+                current_gpu = None
+
+    except FileNotFoundError:
+        logging.error("nvidia-smi command not found. Ensure NVIDIA drivers are installed.")
+    except subprocess.CalledProcessError as e:
+        logging.error(f"Error while querying NVIDIA GPUs: {e}")
+    except Exception as e:
+        logging.error(f"An unexpected error occurred: {e}")
+
+
+def check_bar1_size():
+    """
+    Checks the BAR1 size of all NVIDIA GPUs using nvidia-smi.
+    Logs the BAR1 size for each GPU and ensures it is non-zero.
+    """
+    try:
+        # Run nvidia-smi to get BAR1 memory information
+        result = subprocess.run(
+            ["nvidia-smi", "-q", "-d", "MEMORY"], capture_output=True, text=True, check=True
+        )
+
+        # Parse the output of nvidia-smi
+        output = result.stdout.splitlines()
+
+        current_gpu = None
+        bar1_total = None
+
+        for i, line in enumerate(output):
+            line = line.strip()
+
+            # Detect GPU identifier
+            if line.startswith("GPU"):
+                current_gpu = line.split()[1].strip(":")
+                bar1_total = None
+
+            # Parse BAR1 total size under "BAR1 Memory Usage" section
+            elif "BAR1 Memory Usage" in line:
+                continue  # Skip the header for BAR1 Memory Usage section
+            elif "Total" in line and "BAR1" in output[i - 1]:
+                bar1_total_str = line.split(":")[1].strip()
+                if "MiB" in bar1_total_str:
+                    bar1_total = int(bar1_total_str.split()[0])
+
+            # Once BAR1 size is found, log it
+            if current_gpu is not None and bar1_total is not None:
+                if bar1_total > 1024:
+                    logging.info(f"GPU {current_gpu}: BAR1 size is {bar1_total} MiB.")
+                else:
+                    logging.warning(
+                        f"GPU {current_gpu}: BAR1 size is {bar1_total} MiB. This may indicate an issue."
```

Comment:
Would be nice to update logs to indicate the threshold, like is done for clocks, MRRS (quantitative) or governor and PIX/PXB (qualitative).

Reply:
There's no way to know what the larger BAR1 size is since it varies by GPU

Reply:
😭